server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetStoredScriptAction.java

```diff
@@ -66,19 +65,8 @@ public RestResponse buildResponse(GetStoredScriptResponse response, XContentBuil
 
                 StoredScriptSource source = response.getSource();
                 boolean found = source != null;
-                builder.field(FOUND_PARSE_FIELD.getPreferredName(), found);
 
-                if (found) {
-                    builder.startObject(StoredScriptSource.SCRIPT_PARSE_FIELD.getPreferredName());
-                    builder.field(StoredScriptSource.LANG_PARSE_FIELD.getPreferredName(), source.getLang());
-                    builder.field(StoredScriptSource.SOURCE_PARSE_FIELD.getPreferredName(), source.getSource());
-
-                    if (source.getOptions().isEmpty() == false) {
-                        builder.field(StoredScriptSource.OPTIONS_PARSE_FIELD.getPreferredName(), source.getOptions());
-                    }
-
-                    builder.endObject();
-                }
+                response.toXContent(builder, channel.request());
```

Comment:
I think you can make the response a `StatusToXContentObject` and then use `RestStatusToXContentListener` here.

Reply:
`StatusToXContentObject` starts to conflict with `ToXContentFragment`

Reply:
and `RestStatusToXContentListener` does not work with fragment - `RestGetStoredScriptAction` as well adds `id` into response

Reply:
As I said above, this probably should not be a fragment, I don't get why it would need to be.

Reply:
there is a manual serialization performed in `RestGetStoredScriptAction` - it adds as well `id` of `script` in front of `GetStoredScriptResponse` - agreed to move `id` within `GetStoredScriptResponse`, make it `StatusToXContentObject` and drop `fragment`